### ISSUE TITLE
Disability Rating - Replace On This Page imposter component

### ISSUE THREAD

**Issue by bacitracin** [department-of-veterans-affairs/va.gov-team#115349]
## Issue
OMB's 508 findings for the [Your Disability Ratings](https://staging.va.gov/disability/view-disability-rating/rating) page found that there was no focus indicator on the On This Page links. Closer inspection found that it's an "imposter component" - we should be using the Design System's "On this page" component instead here. Imposter components in our code are considered launch blocking, so we should be replacing it. The DS component happens to include the needed focus styling as well.

<img width="1060" height="310" alt="Image" src="https://github.com/user-attachments/assets/a031d27a-ccb4-4c56-b7e7-6ddf70eeb998" />

## Acceptance Criteria
- [ ]  imposter component is replaced

## Resources
[Design System On This Page component ](https://design.va.gov/components/on-this-page)
[Slack thread with background](https://dsva.slack.com/archives/C08PBUZGP0Q/p1753711903057719)
